resources/assets/js/features/tripsList/components/TripsListItem.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux';
+
+import { showModal } from "../actions";
+import MapDirection from './MapDirection';
+import './css/TripsListItem.scss';
+
+class TripsListItem extends Component {
+    constructor(props){
+        super(props);
+        this.showModal = this.showModal.bind(this);
```

Comment:
don't use global state for modal show/hide, reuse already implemented Modal component that uses local state